feat: Reselect existing contributions when updating a user

chrisinajar opened the pull request:
When you choose to update an existing contributor it now populates the initial selection with their current values, if any.

machour commented on cli.js:
```diff
@@ -56,7 +56,7 @@ function addContribution(argv) {
   var contributions = argv._[2];
   // Add or update contributor in the config file
-  return updateContributors(argv, username, contributions)
+  return updateContributors(argv, username, contributions, argv.contributors)
```
We're already passing `argv` as the first parameter.
Is this really needed?

machour commented on lib/contributors/prompt.js:
```diff
@@ -18,7 +18,7 @@ var contributionChoices = _.flow(
 );
 
-function getQuestions(options, username, contributions) {
+function getQuestions(options, username, contributions, allContributions) {
```
should be named `allContributors` instead (but it can be accessed from `options.contributors`)

machour commented on lib/contributors/prompt.js:
```diff
@@ -29,17 +29,23 @@ function getQuestions(options, username, contributions) {
+      // default values for contributions when updating existing users
+      return allContributions
+        .filter((entry) => entry.login.toLowerCase() === answers.username.toLowerCase())
+        .reduce((memo, entry) => memo.concat(entry.contributions), []);
```
What does `memo` stands for?

kentcdodds replied:
It's a fairly common argument name for the first argument in a reducer.

machour replied:
Thank you for the explanation! Good for me as well then :)

kentcdodds commented:
I'm good to defer to @machour on this :+1:

chrisinajar commented:
Fixed the usability case, removed the dumb extra param (didn't realize options and argv were the same thing while working backwards). I also found an error when it's launched by cli `cli.js add name` as opposed to using the interactive options, fixed that case too.

machour commented:
@chrisinajar thank you for the update.

I re-tested the PR and came across an unhandled case: removing a contribution type doesn't work.

I think this never worked as previously contributions weren't re-selected and were kept anyways, but it just more clear with this PR.

Do you want to fix it here, or do we keep log that in a new issue and merge this baby in as is?

chrisinajar commented:
That's actually an interesting usability issue. There's currently no way to remove accidental contribution types added to the list.

If we changed this to set (instead of merge) the contributions, then direct cli usage (`cli.js add chrisinajar code`) would remove their other contribution types, so I think we'd only want it to behave this way when you went through the interactive UI... which is also weird.

I guess this change just brings to light an awkward usability interaction.

machour commented:
Version [4.6.0](https://github.com/jfmengels/all-contributors-cli/releases/tag/v4.6.0) is out with your contribution 🎉
